lib/msf/core/post/file.rb

```diff
@@ -322,6 +325,148 @@ def exist?(path)
 
   alias exists? exist?
 
+  def grab_icacls_groups(input_string)
+    final_groups_array = []
+    icacls_groups_array = input_string.scan(/^.+? (.+?\\.+?):/)
+    for i in 0...icacls_groups_array.length
+      final_groups_array << icacls_groups_array[i][0].strip # Strip any leading or trailing characters off of each of the groups found via the regex.
+    end
+    final_groups_array
+  end
+
+  def grab_icacls_groups_permissions(input_string)
+    final_permissions_array = []
+    icacls_groups_permissions_array = input_string.scan(/\w:(\(.*\))$/)
+    for i in 0...icacls_groups_permissions_array.length
+      icacls_groups_permissions_array[i][0].strip! # Strip any leading or trailing characters off of each of the group permissions found via the regex.
+
+      # Permissions can exist inside a () section as comma delimited entries. To solve this we first remove the ( and ) characters,
+      # then we split the list on the delimiter, aka the comma character, before finally appending the resulting array to the end of
+      # final_permissions_array.
+      final_permissions_array << icacls_groups_permissions_array[i][0].gsub(/[()]/, '').split(',')
+    end
+    final_permissions_array
+  end
+
+  def win_find_current_user_groups
+    if session.platform != 'windows'
+      print_error('win_find_current_user_groups called on a non-Windows target!')
+      return nil
+    end
+
+    cuser_groups = cmd_exec('whoami /groups').to_s
+    cuser_groups_array = cuser_groups.scan(/^(?:\w+ ?[\w\-]*+ ?[\w\-]*+\\?[\w\-]*+ ?[\w\-]*+ ?[\w\-]*+ ?)/)
+    cuser_groups_length = cuser_groups_array.length
+    for i in 0...cuser_groups_length
+      cuser_groups_array[i].strip! # Strip any leading or trailing characters off the groups captured by the regex.
+    end
+
+    # Remove erronious entries that don't belong here but are captured by the regex.
+    cuser_groups_array.delete('GROUP INFORMATION')
+    cuser_groups_array.delete('Group Name')
+    cuser_groups_array
+  end
+
+  #
+  # See if +path+ on the remote system is accessible in the manner specified by +perm+ by the current user.
+  #
+  # @param path [String] Remote path to check
+  # @param perms [Array] Array of permissions to check for
+  #
+  # @return [Boolean] true if +path+ is accessible to the current user via any of the manners listed in +perms+, otherwise false.
+  #
+  def check_win_path_permissions(path, perms)
+    # First check if the target is Windows. If it is not, we shouldn't continue and should
+    # instead raise an error since this method is designed to only support Windows targets.
+    if session.platform != 'windows'
+      print_error('Check for path permissions attempted on a non-Windows target!')
+      return nil
+    end
```

Comment:
This seems like overkill, this and all `*_win_*` method added here could be be marked private as exposing OS specific methods defeats the purpose of the post/file API.  All calls to this method are also currently guarded with platform check logic.

Reply:
Sounds good, I wasn't sure and since this is library code I wanted to err on the side of caution but agree with your comments here, let me get that fixed.

Reply:
Should be resolved in https://github.com/rapid7/metasploit-framework/pull/15987/commits/38db76c61dbe1f7ff57aa6f7ba0f851366da7a08